stbt_auto_selftest.py

```diff
@@ -147,6 +147,20 @@ def init_worker():
     signal.signal(signal.SIGINT, signal.SIG_IGN)
 
 
+def iterate_with_progress(iterator, width=20, stream=sys.stderr):
+    stream.write('\n')
+    total = len(iterator)
+    for n, v in enumerate(iterator):
+        if n == total:
+            break
+        progress = (n * width) // total
+        stream.write(
+            '\033[K[%s] %8d / %d - Processing %s\r' % (
```

Comment:
It's [ANSI] Erase Line.  I've made this clearer in my fixup.

[ANSI]: https://en.wikipedia.org/wiki/ANSI_escape_code